internal/provider/resource_gitlab_project_environment.go

```diff
@@ -0,0 +1,174 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+// https://docs.gitlab.com/ee/ci/environments/protected_environments.html
+var _ = registerResource("gitlab_project_environment", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_project_environment`" + ` resource you to create and manage an environment in your GitLab project`,
```

Comment:
Please use the doc linking pattern, like:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/49db985b79f667da9244a57d9ff4f107f337b506/internal/provider/resource_gitlab_project_issue.go#L28

and you can remove the comment on line [15](https://github.com/gitlabhq/terraform-provider-gitlab/pull/938/files#diff-5394176d04010b46818ee6dbfbb9d8abfb0fe110ff0c134522cf5fc28630ad51R15)

Reply:
same for the other resource ;)